File: 2018/Javascript/day-1.js
function part1(input) {
  let total = 0;
  for (let i = 0; i < input.length; i++) {
    const current = input[i];
    total += parseInt(current);
  }

  return total;
}

function part2(input) {
  let totals = {};
  let total = 0;

  while (true) {
    for (let i = 0; i < input.length; i++) {
      const current = input[i];
      total += parseInt(current);

      if (totals[total]) {
        return total;
      }

      totals[total] = true;
    }
  }
}

File: 2018/C/Day-1.cs
using System;
using System.Collections.Generic;

class Day1
{
    public int GetFrequency(string[] input)
    {
        int total = 0;
        foreach (string frequency in input)
        {
            total += Convert.ToInt32(frequency);
        }

        return total;
    }

    public int ReachesSameFrequency(string[] input)
    {
        HashSet<int> totals = new HashSet<int>();
        int total = 0;

        while (true)
        {
            foreach (string frequency in input)
            {
                total += Convert.ToInt32(frequency);
                if (totals.Contains(total)) 
                {
                    return total;
                }
                totals.Add(total);
            }
        }
    }
}
